### ISSUE TITLE
rust-playground build fails whole running 'cargo fetch'

### ISSUE THREAD

**Issue by 0x4ndy** [rust-lang/rust-playground#923]
I've been trying to run the build in order to test my changes, using the `compiler/build.sh` script but it keeps on failing at:
```bash
sudo ./build.sh
```
```
...
 => CACHED [toolchain 9/9] ADD --chown=playground entrypoint.sh /playground/tools/                                                                         0.0s
 => CACHED [bare-sources 1/4] RUN cargo init /playground                                                                                                   0.0s
 => CACHED [bare-sources 2/4] ADD --chown=playground Cargo.toml /playground/Cargo.toml                                                                     0.0s
 => CACHED [bare-sources 3/4] ADD --chown=playground crate-information.json /playground/crate-information.json                                             0.0s
 => ERROR [bare-sources 4/4] RUN cargo fetch                                                                                                              33.9s
------
 > [bare-sources 4/4] RUN cargo fetch:
#0 0.485     Updating crates.io index
#0 33.87 error: failed to select a version for the requirement `h2 = "=0.3.17"`
#0 33.87 candidate versions found which didn't match: 0.3.18, 0.3.16, 0.3.15, ...
#0 33.87 location searched: crates.io index
#0 33.87 required by package `playground v0.0.1 (/playground)`
------
Dockerfile:46
--------------------
  44 |     ADD --chown=playground Cargo.toml /playground/Cargo.toml
  45 |     ADD --chown=playground crate-information.json /playground/crate-information.json
  46 | >>> RUN cargo fetch
  47 |
  48 |     # Build our tool for modifying Cargo.toml at runtime
--------------------
ERROR: failed to solve: process "/bin/sh -c cargo fetch" did not complete successfully: exit code: 101
```

Changing the `h2` dependency in `compiler/base/Cargo.toml` to version `0.3.18` (as per output suggestion) resolved the issue for me. However, this might not be ideal solution, and in case it is valid, a more extensive testing would be needed.

**Comment by shepmaster** [comment]
This [isn't just you](https://github.com/rust-lang/rust-playground/actions/runs/4738879057) and is basically another case of #535. The right solution is indeed to update all the dependencies and I can get around to that soon.

> I've been trying to run the build in order to test my changes

For local development, I encourage fetching the existing images when possible. Since you are changing the containers, that wouldn't make sense. In those cases, I usually manually edit the `compiler/base/Cargo.toml` file to remove all but one or two dependencies (I often pick `tokio` and `itertools`). This makes the builds much faster. You can also set the appropriate environment variables to build just one channel (e.g. `stable`) and skip building the tool containers.

**Comment by 0x4ndy** [comment]
Oh, ok, gotcha. I have done the change and successfully build it already, so if you want I can push it to my fork and make a pull request.

Speaking of the local development, is there a procedure/guide about how to actually deploy/run the Rust Playground? I wanted to test my change #626.

**Comment by shepmaster** [comment]
> push it to my fork and make a pull request.

Sure!

> is there a procedure/guide about how to actually deploy/run the Rust Playground locally

No, but there really should be.

**Comment by shepmaster** [comment]
Original issue should be fixed by #924
